kinks/level_4/introduction_to_programming_i/README.md

```diff
@@ -123,15 +129,50 @@ Note that the `distance`/`km`/`miles_travelled` property is increasing towards t
+#### 801, All stages
+More spotlight stages, which may cause Firefox users to not see the spotlights. 
+To get it to work, comment out `blendMode(DARKEST)` during your work and to uncomment again before submission. Or don't.
+
+Tips:
+- The 1st seat in the 1st row will be `seats[0][0]` 
+- Rows and seats relate to `seats[row][seat]` and not vice versa
+- You don't need to factor in reverse counting of rows such that front to back make row numbers increase like in a real theatre. For this case the first row is at the back of the theatre (i.e. the top of the screen)  
+
+#### 801, Stage 3-4
+- Many students waste time here trying to spot patterns in the audience members based on their appearance. The information about the seat numbers of gang/perpetrators is already defined in an array. It appears that any information given about appearance (hats, retro glasses, etc.) is merely to confirm that you have used the array to traverse the seats in the correct manner and not to actually identify gang members.
+
+#### 801, Stage 4
+Here the challenge is to traverse a two-dimensional array using a normal (one-dimensional) array. There are different ways to approach this, but these work:
+
+- create additional variables and use a nested-for with conditionals, or
+- convert the 1D array to a 2D array and compare against the 2D array
+
+It appears that the grader is looking for two simple nested for-loops in the solution, so more elegant ways involving math to convert your 1D iterator into two 2D co-ordinates does not work, even when all gang members are correctly identified. This, for example, was not accepted by the grader: 
+`for (var i = 0; i < suspect.length; i++)
+operaFolks[floor(i/10)][i%10].recognised = (suspect[i]);`
+
+#### 802, Stage 1
+Tip: You can comment out the line `if (frameCount % 7 == 0)` in order to speed up the deck spread sequence. It might save some time during testing. 
+
+#### 802, Stage 2
 ##### Random function
 _"They ask you to create an array of random integers between two values. The `random()` function in `p5.js` includes the lower number, but excludes the upper number. So if you do `random(5,9)`, it might well generate a **5**, but never a **9**._
 
 _Naturally I incremented the upper value by one so it'd actually generate values between the two numbers requested, but the puzzle actually expects you to just naively slot the two values into random and not fix the bug."_ (reported by *Pehoulihan*)
 
+#### 802, Stage 3
+Tip: When comparing cards, you can't simply compare objects against each other as the objects may be structured differently. Rather consider the properties of the different objects in the arrays and compare similar properties against one another.
+
+#### 802, Stage 4
+Take a good look at some form of documentation to understand how the JavaScript `splice()` function works. [MDN `Array.splice` documentation] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
```

Comment:
This link will not render properly: [MDN `Array.splice` documentation] (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)

The extra space needs to be removed before the left parenthesis:
[MDN `Array.splice` documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)

Reply:
Fixed.